source/slang/slang-ir-entry-point-uniforms.cpp

```diff
@@ -309,6 +329,13 @@ struct CollectEntryPointUniformParams : PerEntryPointPass
             if (isVaryingParameter(paramLayout))
                 continue;
 
+            // For Metal targets, skip collection of StructuredBuffer parameters
+            // so they remain as direct entry point parameters and can generate
+            // proper [[buffer(N)]] attributes
+            //
+            if (shouldSkipStructuredBufferCollectionForMetal(param, m_options))
```

Comment:
What is special about structured buffers as opposed to other resource types? Won't textures etc. have the same issue and also need to be skipped? We should clean up the code here into a single `shouldCollectEntryPointParamInConstantBuffer` function for readbility.

I think for metal you may want to skip all resource typed parameters from collection.